xwayland/selection/incoming.c

```diff
@@ -450,22 +444,30 @@ int xwm_handle_xfixes_selection_notify(struct wlr_xwm *xwm,
 
 	selection->owner = event->owner;
 
-	// We have to use XCB_TIME_CURRENT_TIME when we claim the
-	// selection, so grab the actual timestamp here so we can
-	// answer TIMESTAMP conversion requests correctly.
+	if (selection->window == event->owner) {
```

Comment:
This new `if` seems to conflict with the following `if`.

Reply:
It does, and I thought I pushed a fix to remove it, but it looks like I didn't.